platformio.ini

```diff
@@ -12,7 +12,7 @@
 platform = espressif8266
 framework = arduino
 board_f_cpu = 160000000L
-build_flags = -Wall
+build_flags = !python scripts/firmware_version.py -Wall
```

Comment:
[No need for a python script.](http://docs.platformio.org/en/latest/projectconf/section_env_build.html#dynamic-build-flags) Please change to:
```ini
build_flags = !echo "-DFIRMWARE_VERSION="$(git describe --abbrev=8 --dirty --always --tags)
```
and remove `firmware_version.py`.

Reply:
Have you tested that this gives tags for tagged versions?

I would leave a script for flexibility.

Reply:
Yes, that is  `--tags` for.